src/main/java/com/vmware/vim25/ws/WSClient.java

```diff
@@ -68,7 +73,18 @@ public WSClient(String serverUrl, boolean ignoreCert) throws MalformedURLExcepti
         log.trace("Creating WSClient to server URL: " + serverUrl);
         log.trace("Ignore ssl: " + ignoreCert);
         this.baseUrl = new URL(serverUrl);
-        this.ignoreCert = ignoreCert;
+        
+        try {
+            this.sslSocketFactory = ignoreCert ? TrustAllSSL.getTrustContext().getSocketFactory() : null;
+        } 
+        catch (NoSuchAlgorithmException e) {
+            log.debug("Unable to find suitable algorithm while attempting to communicate with remote server.", e);
+            throw new IllegalArgumentException("Unable to find suitable algorithm while attempting to communicate with remote server.", e);

```

Comment:
Note: Adding this means that you should add ```IllegalArgumentException``` to the ```throws``` clause, as a best-practice measure.